routing/matcher.go

```diff
@@ -358,8 +376,10 @@ func (m *matcher) match(r *http.Request) (*Route, map[string]string) {
 		path = path[:len(path)-1]
 	}
 
+	lrm := &leafRequestMatcher{r, path}
+
 	// first match fixed and wildcard paths
-	leaves, params := matchPathTree(m.paths, path)
+	leaves, params := matchPathTree(m.paths, path, lrm)
 	l := matchLeaves(leaves, r, path)
```

Comment:
maybe we can change the pathmux matcher a bit, so that it would return not only a bool but also the matched value.